### PR TITLE
Add documentation and cleanups for PlanningRequestAdapter and PlanningRequestAdapterChain classes

### PR DESCRIPTION
**Pull request by sjahr**
### Description

- Documentation & Cleanups for PlanningRequestAdapter & PlanningRequestAdapterChain classes

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit2/blob/main/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://moveit.picknik.ai/humble/doc/examples/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit2/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Review by henningkayser — COMMENTED** [review]
I approve these changes in general. But why did you mix in the fixes for MSA and PSM, they don't seem related at all?

File: moveit_setup_assistant/moveit_setup_framework/include/moveit_setup_framework/templates.hpp [hunk 1]
```diff
@@ -69,7 +69,7 @@ class TemplatedGeneratedFile : public GeneratedFile
 
   bool write() override;
 
-  static std::vector<TemplateVariable> variables_;
+  static std::vector<TemplateVariable> variables;
```
Comment (henningkayser) [inline moveit_setup_assistant/moveit_setup_framework/include/moveit_setup_framework/templates.hpp:72]: are we sure that we want to drop `_` for static member variables? I've just found the same issue here https://github.com/ros-planning/moveit2/actions/runs/4862228141/jobs/8670864807?pr=2119#step:12:9677, but I'm not convinced that it's actually a good format rule

Comment (sjahr) [reply moveit_setup_assistant/moveit_setup_framework/include/moveit_setup_framework/templates.hpp:72]: I don't have a strong opinion about this so I can definitely revert it, but it is interesting that this CI error seems to pop up randomly. This PR and the other did not touch the files where it occurs

**Comment by sjahr** [comment]
@henningkayser I cleaned it up

**Pull request merged**